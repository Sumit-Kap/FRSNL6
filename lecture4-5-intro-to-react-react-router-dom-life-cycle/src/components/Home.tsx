import React from "react";
import { Link, RouteComponentProps, withRouter } from "react-router-dom";
interface Props {}
interface ResponseData {
  id: number;
  price: number;
  description: string;
  image: string;
}
interface State {
  response: ResponseData[];
  showOtherData: boolean;
}
class Home extends React.PureComponent<RouteComponentProps<Props>, State> {
  constructor(props: RouteComponentProps<Props>) {
    super(props);
    // console.log()
    this.state = {
      response: [],
      showOtherData: false,
    };
  }
  getProductsData = async () => {
    const apiResponse = await fetch("https://fakestoreapi.com/products/");
    const responseData = await apiResponse.json();
    this.setState({
      response: responseData,
    });
    console.log("print data", responseData);
  };
  componentDidMount() {
    this.getProductsData();
  }
  render() {
    const { response } = this.state;
    // console.log("print ", response);
    if (response.length === 0) {
      return (
        <>
          <h1>Hello</h1>
          <div className="loader">Loader........</div>
        </>
      );
    }
    return (
      <div>
        {/* <input type="text" data-testid="input" /> */}
        <div className="product-list">
          {response.map((resp, index) => (
            <div
              className="product-cards"
              data-test-id={index}
              onClick={() => {
                // this.setState({
                //   showOtherData: true,
                // });
                // this.props.history.push(`products/${resp.id}`);
              }}
            >
              <img src={resp.image} />
              <div className="product-description">{resp.description}</div>
              <div className="product-price">{resp.price}</div>
            </div>
          ))}
        </div>
      </div>
    );
  }
}

export default Home;
