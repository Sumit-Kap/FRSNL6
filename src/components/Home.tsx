import React from "react";
interface Props {}
interface ResponseData {
  id: number;
  price: number;
  description: string;
  image: string;
}
interface State {
  response: ResponseData[];
}
export default class Home extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      response: [],
    };
  }
  getProductsData = async () => {
    const apiResponse = await fetch("https://fakestoreapi.com/products/");
    const responseData = await apiResponse.json();
    this.setState({
      response: responseData,
    });
    console.log("print data", responseData);
  };
  componentDidMount() {
    this.getProductsData();
  }
  render() {
    const { response } = this.state;

    if (response.length === 0) {
      return <div className="loader">Loader........</div>;
    }
    return (
      <div>
        <div className="product-list">
          {response.map((resp) => (
            <div className="product-cards">
              <img src={resp.image} />
              <div className="product-description">{resp.description}</div>
              <div className="product-price">{resp.price}</div>
            </div>
          ))}
        </div>
      </div>
    );
  }
}
