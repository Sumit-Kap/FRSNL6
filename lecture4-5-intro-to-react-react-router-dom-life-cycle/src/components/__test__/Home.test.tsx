import {
  render,
  screen,
  fireEvent,
  waitFor,
  cleanup,
} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import Home from "../Home";
import products from "../__mocks__/products";

describe("Home Test Scenario", () => {
  beforeEach(() => {
    // jest.mock(gloa)
    global.fetch = () => {
      return Promise.resolve({
        json: () => products.getProducts(),
      });
    };
  });
  afterEach(cleanup);
  it("Look for loading bar in the component", async () => {
    const { getByText, getByTestId, container } = render(<Home />);

    expect(getByText("Loader........")).toBeDefined();
    // waitFor(expect(get("input"));
    waitFor(() => expect(getByTestId("input")).toBeDefined);
    waitFor(() => expect(getByTestId(0)).toBeDefined);
    // removeDynamicAttirbutes();
    expect(container).toMatchSnapshot();
  });
});
