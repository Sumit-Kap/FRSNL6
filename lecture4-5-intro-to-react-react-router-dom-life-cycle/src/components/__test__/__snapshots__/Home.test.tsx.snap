// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Test Scenario Look for loading bar in the component 1`] = `
<div>
  <h1>
    Hello
  </h1>
  <div
    class="loader"
  >
    Loader........
  </div>
</div>
`;
